### ISSUE TITLE
No notification

### ISSUE THREAD

**Issue by dordzhiev** [Dev-hwang/flutter_foreground_task#136]
I do not remember what changes I went through, but on android 13 (only) the foreground task does not request access to notifications as before.

**AndroidManifest.xml**
`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.application">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
    <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" />

    <application
        android:name="${applicationName}"
        android:enableOnBackInvokedCallback="true"
        android:icon="@mipmap/ic_launcher"
        android:label="Example App">
        <service
            android:name="com.pravera.flutter_foreground_task.service.ForegroundService"
            android:stopWithTask="true" />
        <activity
            android:name=".MainActivity"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:exported="true"
            android:hardwareAccelerated="true"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:windowSoftInputMode="adjustResize">
            
            <meta-data
                android:name="io.flutter.embedding.android.NormalTheme"
                android:resource="@style/NormalTheme" />
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>
`

**Comment by dordzhiev** [comment]
**Handler**
```
import 'dart:async';
import 'dart:isolate';

import 'package:flutter_foreground_task/flutter_foreground_task.dart';
import 'package:geolocator/geolocator.dart';
import 'package:graphql_flutter/graphql_flutter.dart';
import 'package:intl/intl.dart';

import '../../locator.dart';
import '../../shared/services/api_service.dart';
import '../../shared/services/geolocator_service.dart';

const startedKey = 'started';

class FirstTaskHandler extends TaskHandler {
  SendPort? _sendPort;
  bool? started;
  StreamSubscription? streamSubscription;

  @override
  Future<void> onStart(DateTime timestamp, SendPort? sendPort) async {
    await initHiveForFlutter();
    await setupLocator();
    _sendPort = sendPort;
    started = await FlutterForegroundTask.getData(key: startedKey);
    if (!(started ?? false)) {
      await APIService.startCourierSession(
        await GeolocatorService.getPosition(),
      );
    }
  }

  @override
  Future<void> onEvent(DateTime timestamp, SendPort? sendPort) async {
    await FlutterForegroundTask.updateService(
      notificationTitle: 'Вы на линии',
      notificationText:
          'Передаём геопозицию.. (${DateFormat('hh:mm:ss').format(timestamp)})',
    );
    await APIService.updateCurrentLocation(
      await GeolocatorService.getCurrentPosition(),
    );
  }

  @override
  void onButtonPressed(String id) {}

  @override
  Future<void> onDestroy(DateTime timestamp, SendPort? sendPort) async {
    await FlutterForegroundTask.clearAllData();
    await streamSubscription?.cancel();
  }
}
```
**Service**
```
import 'dart:isolate';

import 'package:flutter_foreground_task/flutter_foreground_task.dart';

import '../../main.dart';

class ForegroundService {
  static void _initForegroundTask() {
    FlutterForegroundTask.init(
      androidNotificationOptions: AndroidNotificationOptions(
        channelId: 'notification_channel_id',
        channelName: 'Foreground Notification',
        channelDescription:
            'This notification appears when the foreground service is running.',
        channelImportance: NotificationChannelImportance.LOW,
        priority: NotificationPriority.LOW,
        iconData: const NotificationIconData(
          resType: ResourceType.mipmap,
          resPrefix: ResourcePrefix.ic,
          name: 'launcher',
        ),
        buttons: [
          const NotificationButton(id: 'stopSession', text: 'Остановить'),
        ],
      ),
      iosNotificationOptions: const IOSNotificationOptions(
        showNotification: true,
        playSound: false,
      ),
      foregroundTaskOptions: const ForegroundTaskOptions(
        interval: 5000,
        isOnceEvent: false,
        autoRunOnBoot: true,
        allowWakeLock: true,
        allowWifiLock: true,
      ),
    );
  }

  static Future<ReceivePort?> startForegroundTask() async {
    _initForegroundTask();
    if (await FlutterForegroundTask.isRunningService) {
      await FlutterForegroundTask.stopService();
    }
    bool reqResult = await FlutterForegroundTask.startService(
      notificationTitle: 'Foreground Service is running',
      notificationText: 'Tap to return to the app',
      callback: startCallback,
    );
    return reqResult ? FlutterForegroundTask.receivePort : null;
 }
}
```

**Comment by blopker** [comment]
See the fix in #142
